src/core/utils/constants.ts

```diff
@@ -12,3 +12,8 @@ export enum PermissionMode {
   LibraryRead = 'library_read',
   LibraryWrite = 'library_write',
 }
+
+export enum PrincipalType {
+  User = 'user',
+  Group = 'group',
```

Comment:
Should we use users and groups here? There is users and groups in principles structure and I think it would be better that we use one consistent key for principalType.

Reply:
But `PrincipalType` here is used to indicate the type of one principal, like this: ` {type:'user',name:'user1',permissions:['read']}`, use plural maybe weird.

Reply:
Yes this is only used internally through devs, users won't be able to see the type here. If we decide to use different enums in this case, we'd better declare a new enum to indicate the principalTypes in principal structure.